raiden/messages.py

```diff
@@ -1065,6 +1091,11 @@ def reward_proof_signature(self) -> Optional[Signature]:
 
     def _data_to_sign(self) -> bytes:
         """ Return the binary data to be/which was signed """
+        if self.non_closing_signature is None:
+            raise ValueError("non_closing_signature missing, did you forget to sign()?")
+        if self.reward_proof_signature is None:
```

Comment:
something is not right here, this method is called by `sign` and that docstring says `[it creates] the `reward_proof_signature` for the monitoring request`

Reply:
And it does, `reward_proof_signature` is [a property](https://github.com/raiden-network/raiden/blob/75c3a8ba884887010d2c2b8167410b837133364c/raiden/messages.py#L1088-L1090) that returns the property `signature`, which is computed by [sign](https://github.com/raiden-network/raiden/blob/75c3a8ba884887010d2c2b8167410b837133364c/raiden/messages.py#L1119) after calling [this function](https://github.com/raiden-network/raiden/blob/75c3a8ba884887010d2c2b8167410b837133364c/raiden/messages.py#L1118)

Reply:
Yes, the check for reward_proof_signature does not make sense. The only reason why it didn't stop the tests is because the signature is initialized with `EMPTY_SIGNATURE` instead of `None`. I don't see a good place to put that check, so I'll just remove it.